tests/integration_tests/test_hub_state_monitor.py

```diff
@@ -90,3 +90,75 @@ def test_on_connect(self):
             on_connect.assert_called_once_with(monitor.connected_socket)
         finally:
             monitor.stop()
+
+    def test_round_trip_latency(self):
+        MESSAGES_TO_SEND = 100
+        connected = threading.Event()
+        connected.clear()
+
+        self.recvd_all = threading.Event()
+        self.recvd_all.clear()
+
+        self.recv_count = 0
+        self.recv_latencies = []
+
+        def on_state_update(
```

Comment:
a reminder here that this is running on a different thread than the test itself might add clarity